sprite.h

```diff
@@ -14,9 +14,13 @@
 #include "context.h"
 #include "graphics.h"
 
+class Trail;
+
 class Sprite : public Identifiable<Empty> {
 public:
-	Sprite(const Texture *texture, const Point &home);
+	Sprite(const Texture *texture, const Point &home, const bool has_trail = true);
+
+	std::vector<Trail> trail;
```

Comment:
This should be protected rather than public (I know why you made it public, but the next comment describes why it doesn't have to be), and nitpick: as a field it should be called `m_trail` instead of just `trail`.

Reply:
I have a feeling that private, protected, and public are forever gonna be one of those things that you **have** to tell me to use properly otherwise I just won't.